File: ai/services/analyst_service.py
from agents.analyst.schemas import (AnalysisResult)

VALID_RISKS = {
    "LOW",
    "MEDIUM",
    "HIGH",
    "CRITICAL"
}

class AnalystService:

    @staticmethod
    def process(text: str):

        result = AnalystAgent.analyze(text)

        risk = str(
            result.get(
                "riskLevel",
                "MEDIUM"
            )).upper()

        if risk not in VALID_RISKS:
            risk = "MEDIUM"

        result["riskLevel"] = risk

        validated = AnalysisResult(**result)

        return validated.model_dump()

File: backend/src/ai/analyst/analyst.service.ts
import { AnalystAgent } from "./analyst.agent.js";

import { AnalystSchema } from "./analyst.schema.js";

export class AnalystService {
  static async process(text: string) {
    const result = await AnalystAgent.analyze(text);

    return AnalystSchema.parse(result);
  }
}
